### PR TITLE
Prevent index ambiguity

### PR DESCRIPTION
**Pull request by alpe**
We have currently a multi-key and a unique key index. Calling it Index in a function is not very clear.

### CODE REVIEW

File: incubator/orm/index.go [hunk 1]
```diff
@@ -27,12 +27,12 @@ type MultiKeyIndex struct {
 	indexer   indexer
 }
 
-func NewIndex(builder Indexable, prefix byte, indexer IndexerFunc) *MultiKeyIndex {
+func NewMultiKeyIndex(builder Indexable, prefix byte, indexer IndexerFunc) *MultiKeyIndex {
```
Comment (aaronc) [inline incubator/orm/index.go]: If we go ahead with this we should also rename `IndexerFunc` to `MultiKeyIndexerFunc`

**Review by aaronc — COMMENTED** [review]
Do you think that this renaming actually makes things clearer @alpe? As I said it was a minor quibble and looking how it affects the naming in `example_test.go` I'm thinking maybe `NewIndex` was clear enough and definitely less verbose. So maybe I was wrong about unique index being the more common case and we should just keep it as you had it?

**Comment by alpe** [comment]
I don't have a strong opinion on the naming. With this implementation it is well defined what kind of indexes we have. There is no "default" Index type that may give a wrong expectation. But we can always add it later when it turns out that it is hardly missed by the people. :-)
Changing `NewIndex` from multi to unique is a breaking change on the other hand.

**Comment by aaronc** [comment]
It's a small issue but also naming is one of the hardest things in CS right? Maybe we discuss quickly when we're live on Thursday.

**Comment by alpe** [comment]
@aaronc can I decide on the naming? I am ok with any of the two options.

**Comment by aaronc** [comment]
Let's keep it as is for now thanks.

**Pull request closed without merging**